python/tank_vendor/shotgun_authentication/session_cache.py

```diff
@@ -0,0 +1,209 @@
+            os.remove(info_path)
+            logger.debug("Session cleared.")
+        else:
+            logger.debug("Session file not found: %s", info_path)
+    except:
+        logger.exception("Couldn't delete the session cache file!")
+
+
+def get_session_data(base_url):
+    """
+    Returns the cached login info if found.
+
+    :param base_url: The site we want the login information for.
+
+    :returns: Returns a dictionary with keys login and session_token or None
+    """
+    # Retrieve the location of the cached info
+    info_path = _get_session_data_location(base_url)
+    # Nothing was cached, return an empty dictionary.
+    if not os.path.exists(info_path):
+        logger.debug("No cache found at %s" % info_path)
+        return None
+    try:
+        # Read the login information
+        config = SafeConfigParser({"login": None, "session_token": None})
+        config.read(info_path)
+        if not config.has_section("LoginInfo"):
+            logger.debug("No Login info was found")
+            return None
+
+        login = config.get("LoginInfo", "login", raw=True)
+        session_token = config.get("LoginInfo", "session_token", raw=True)
+
+        if not login or not session_token:
+            logger.debug("Incomplete settings (login:%s, session_token:%s)" % (login, session_token))
+            return None
+
+        return {"login": login, "session_token": session_token}
+    except Exception:
+        logger.exception("Exception thrown while loading cached session info.")
+        return None
+
+
+def cache_session_data(host, login, session_token):
+    """
+    Caches the session data for a site and a user.
+
+    :param host: Site we want to cache a session for.
+    :param login: User we want to cache a session for.
+    :param session_token: Session token we want to cache.
+    """
+    # Retrieve the cached info file location from the host
+    info_path = _get_session_data_location(host)
+
+    # make sure the info_dir exists!
+    info_dir, info_file = os.path.split(info_path)
+    if not os.path.exists(info_dir):
+        os.makedirs(info_dir, 0700)
+
+    logger.debug("Caching login info at %s...", info_path)
+    # Create a document with the following format:
+    # [LoginInfo]
+    # login=username
+    # session_token=some_unique_id
+    config = SafeConfigParser()
+    config.add_section("LoginInfo")
+    config.set("LoginInfo", "login", login)
+    config.set("LoginInfo", "session_token", session_token)
+    # Write it to disk.
+    with open(info_path, "w") as configfile:
+        config.write(configfile)
+    logger.debug("Cached!")
+
+
+def generate_session_token(hostname, login, password, http_proxy):
+    """
+    Generates a session token for a given username/password on a given site.
+
+    :param hostname: The host to connect to.
+    :param login: The user to get a session for.
+    :param password: Password for the user.
+    :param http_proxy: Proxy to use. Can be None.
```

Comment:
The reason why I prefer having the parameter explicit is that I think we tend to forget about http proxies because we rarely test using them. Making the parameter not optional forces you to be aware of it.

Originally I defaulted it in a couple of methods to None and then at some point I realized I was forgetting about it and not passing it around, so that's when I made the change.